WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipment Tracking Section/Add Tracking/ManualTrackingViewController.swift

```diff
@@ -448,6 +458,20 @@ private extension ManualTrackingViewController {
     }
 }
 
+// MARK: - UISheetPresentationControllerDelegate comformance
+//
+/// Asks permission to dismiss. We call this delegate method whenever the user attempts to dismiss the View Controller via the pull-down gesture.
+extension ManualTrackingViewController: UISheetPresentationControllerDelegate {
+    func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
+        if hasUnsavedChanges {
```

Comment:
I guess we can move the actual showing the confirmation alert to `presentationControllerDidAttemptToDismiss`, leaving in `presentationControllerShouldDismiss` just the logic of whether the should dismiss or not.